joplin/pages/department_page/tests.py

```diff
@@ -5,24 +5,31 @@
 
 # If we don't have any associated department group
 @pytest.mark.django_db
-def test_department_page_no_department_group():
-    page = DepartmentPageFactory.build(slug="department_slug")
+def test_department_page_no_department_group(home_page):
+    page = DepartmentPageFactory.create(slug="department_slug", parent=home_page)
```

Comment:
All the factories for these URL tests now have to be "created" instead of "built". A merely "created" factory fixture is not saved to the database, and therefore it does not have a parent. Without a parent, we cannot have a janis_publish_url, since the base_url is now being set by the parent.

Reply:
💜